Pixel Game/Assets/Scripts/SceneTransition.cs

```diff
@@ -51,11 +51,8 @@ private void Start()
         
         // Run the Start() function for the scripts that are transfered between scenes
         // since they cant run Start() on scene change by themselves
-        dataToPassBetweenScenes.Start(); 
-        playerHealth.Start();
+        playerHealth.OnSceneChange();
         playerInput.Start();
-        playerAttack.Start();

```

Comment:
Where do we run the removed scripts Start() function?

Reply:
We didn't need to call it on every scene change. Only when it gets references to objects that get destroyed on scene change.

Reply:
Not sure o follow here. So the scripts removed will not run Start() on scenechange?

Reply:
No they won't and there was no need to since they don't get destroyed. I kept `PlayerHealth` since it needs to get a new reference to a health slider because the old one gets destroyed on scene change. It also needs to update the player's spawn position so that he respawns where he entered the scene (maybe this should change in the future though so you only respawn at certain save points).
I also kept `PlayerInput` since it references the `GuiScreenManager`which gets destroyed/recreated on scene change.